### PR TITLE
Enable setting a EurekaClient instance in DiscoveryManager

### PR DESCRIPTION
**Pull request by asibross**

### CODE REVIEW

File: eureka-client/src/main/java/com/netflix/discovery/DiscoveryManager.java [hunk 1]
```diff
@@ -42,6 +42,7 @@
 public class DiscoveryManager {
     private static final Logger logger = LoggerFactory.getLogger(DiscoveryManager.class);
     private DiscoveryClient discoveryClient;
+    private EurekaClient eurekaClient; // Might be a proxy to DiscoveryClient
```
Comment (troshko111) [inline eureka-client/src/main/java/com/netflix/discovery/DiscoveryManager.java:45]: Can you call it something like `clientOverride`?

File: eureka-client/src/main/java/com/netflix/discovery/DiscoveryManager.java [hunk 2]
```diff
@@ -58,6 +59,10 @@ public void setDiscoveryClient(DiscoveryClient discoveryClient) {
         this.discoveryClient = discoveryClient;
     }
 
+    public void setEurekaClient(EurekaClient eurekaClient) {
```
Comment (troshko111) [inline eureka-client/src/main/java/com/netflix/discovery/DiscoveryManager.java:62]: Same here (`setClientOverride`).

**Review by troshko111 — APPROVED** [review]

File: eureka-client/src/main/java/com/netflix/discovery/DiscoveryManager.java (hunk 1 above)
Comment (asibross) [reply eureka-client/src/main/java/com/netflix/discovery/DiscoveryManager.java]: Done

File: eureka-client/src/main/java/com/netflix/discovery/DiscoveryManager.java (hunk 2 above)
Comment (asibross) [reply eureka-client/src/main/java/com/netflix/discovery/DiscoveryManager.java]: Done

**Pull request merged**